build-system/tasks/dist.js

```diff
@@ -231,174 +224,135 @@ async function prebuild() {
   await preBuildWebPushPublisherFiles();
 }
 
-/**
- * Copies the css from the build folder to the dist folder
- * @return {!Promise}
- */
-function copyCss() {
-  const startTime = Date.now();
-
-  cssEntryPoints.forEach(({outCss}) => {
-    fs.copySync(`build/css/${outCss}`, `dist/${outCss}`);
-  });
-
-  return toPromise(
-    gulp
-      .src('build/css/amp-*.css', {base: 'build/css/'})
-      .pipe(gulp.dest('dist/v0'))
-  ).then(() => {
-    endBuildStep('Copied', 'build/css/*.css to dist/v0/*.css', startTime);
-  });
-}
-
 /**
  * Copies parsers from the build folder to the dist folder
- * @return {!Promise}
  */
-function copyParsers() {
+async function copyParsers() {
   const startTime = Date.now();
-  return fs.copy('build/parsers', 'dist/v0').then(() => {
-    endBuildStep('Copied', 'build/parsers/ to dist/v0', startTime);
-  });
+  await fs.copy('build/parsers', 'dist/v0');
+  endBuildStep('Copied', 'build/parsers/ to dist/v0', startTime);
 }
 
 /**
  * Build amp-web-push publisher files HTML page.
- *
- * @return {!Promise<!Array>}
  */
 async function preBuildWebPushPublisherFiles() {
-  mkdirSync('dist');
-  mkdirSync('dist/v0');
-  const promises = [];
-
-  WEB_PUSH_PUBLISHER_VERSIONS.forEach((version) => {
-    WEB_PUSH_PUBLISHER_FILES.forEach((fileName) => {
-      const basePath = `extensions/amp-web-push/${version}/`;
-      const tempBuildDir = `build/all/amp-web-push-${version}/`;
+  for (const version of WEB_PUSH_PUBLISHER_VERSIONS) {
+    for (const fileName of WEB_PUSH_PUBLISHER_FILES) {
+      const srcPath = `extensions/amp-web-push/${version}`;
+      const destPath = `build/all/amp-web-push-${version}`;
+      await fs.ensureDir(destPath);
 
       // Build Helper Frame JS
-      const js = fs.readFileSync(basePath + fileName + '.js', 'utf8');
-      const builtName = fileName + '.js';
-      const promise = toPromise(
-        gulp
-          .src(basePath + '/*.js', {base: basePath})
-          .pipe(file(builtName, js))
-          .pipe(gulp.dest(tempBuildDir))
-      );
-      promises.push(promise);
-    });
-  });
-  return Promise.all(promises);
+      const js = await fs.readFile(`${srcPath}/${fileName}.js`, 'utf8');
+      const builtName = `${fileName}.js`;
+      await fs.writeFile(`${destPath}/${builtName}`, js);
+      const jsFiles = globby.sync(`${srcPath}/*.js`);
+      for (const jsFile of jsFiles) {
+        await fs.copy(jsFile, `${destPath}/${path.basename(jsFile)}`);
```

Comment:
same question re. parallelization vs. waterfalling

Reply:
Same reason as above. One thing to add is that for cases when there genuinely is reason to parallelize (e.g. building 150 extensions or dozens of JS files at once), we do indeed use `Promise.all`.